### PR TITLE
Update visualization howto

### PR DESCRIPTION
**Pull request by matt-gardner**
I _think_ I modified everything I needed to for updating the demo tutorials.  I might have missed something, though - this was a little rushed.

### CODE REVIEW

File: tutorials/getting_started/making_predictions_and_creating_a_demo.md [hunk 1]
```diff
@@ -234,7 +234,12 @@ that the predictor is expecting, or the demo won't work!
 
 ## Customizing the Demo
 
-Our JSON "visualization" is not particularly impressive.
+Our JSON "visualization" is not particularly impressive.  If you want to visualize model internals,
+with easy attention heat maps, you might want to look at our [how-to for visualizing model
+internals](../how_to/visualizing_model_internals.md).  That how-to relies on installing `npm` and
+using react; if you want a simpler way of just using html and javascript (but with less
```
Comment (joelgrus) [inline tutorials/getting_started/making_predictions_and_creating_a_demo.md]: capitalization for HTML and JavaScript

File: tutorials/how_to/visualizing_model_internals.md [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-instructions for running the main demo in the [demo README](../../demo/README.md), but you'll have
-to figure out how to either get the same thing running with your stand-alone demo or add your model
-to what's already there.  I said this was still rough.
-
-You should also be at least vaguely familiar with [react](https://reactjs.org) and how JSX
-components work.
+1. You need to have a text in / text out demo of your model already running using our simple
+   server.  See the [tutorial on creating a
+demo](../getting_started/making_predictions_and_creating_a_demo.md) for how to do this.
+2. We'll be using a react app to generate HTML and javascript for the fancier demo, so you need to
```
Comment (joelgrus) [inline tutorials/how_to/visualizing_model_internals.md]: capitalization on React (elsewhere too)

File: tutorials/how_to/visualizing_model_internals.md [hunk 3]
```diff
@@ -46,35 +50,54 @@ tokenize your data, and pass along the tokens there), but it was easiest this wa
...
+your model expects to receive in `predict_json`.
 
-The `HeatMap` component we used expects a two-dimensional data array.  If you have a
-one-dimensional data array you want to visualize, this is still possible, but you need to do
-something like `attention = attention.map(x => [x])` in the javascript to make your
-one-dimensional array into a two-dimensional array with length 1 on the x axis.  You also need to
-provide a single `xLabel` to the `HeatMap`, like `xLabels={['Probabilities']}`.
+Once you've modified these, you can check that you've done it correctly by clicking "Run" with an
+example and making sure that your simple server receives and can respond to the request.  You
+should be running the simple server on port 8000, and you should see the server log a `POST`
```
Comment (joelgrus) [inline tutorials/how_to/visualizing_model_internals.md]: I think this demo should use whatever the default port is for the simple_server, so either the demo should look for the backend on port 8888 (less work), or the simple_server should default to 8000 (which requires changing other tutorials, probably)

File: tutorials/getting_started/making_predictions_and_creating_a_demo.md (hunk 1 above)
Comment (matt-gardner) [reply tutorials/getting_started/making_predictions_and_creating_a_demo.md]: Done.

File: tutorials/how_to/visualizing_model_internals.md (hunk 2 above)
Comment (matt-gardner) [reply tutorials/how_to/visualizing_model_internals.md]: Done.

File: tutorials/how_to/visualizing_model_internals.md (hunk 3 above)
Comment (matt-gardner) [reply tutorials/how_to/visualizing_model_internals.md]: Yeah, good point.  Any complaints about changing the default port to 8000?

Comment (matt-gardner) [reply tutorials/how_to/visualizing_model_internals.md]: I did a grep through the repo (including tutorials) for `8888`, and there were only two lines to change, so I changed them.

Comment (joelgrus) [reply tutorials/how_to/visualizing_model_internals.md]: 👍

**Pull request merged**